hooks/tk-multi-publish2/photoshopcc.basic/collector.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import sgtk
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class PhotoshopCCSceneCollector(HookBaseClass):
+    """
+    Collector that operates on the photoshop cc active document. Should inherit
+    from the basic collector.
+    """
+
+    def process_current_session(self, parent_item):
+        """
+        Analyzes the open documents in Photoshop and creates publish items
+        parented under the supplied item.
+
+        :param parent_item: Root item instance
+        """
+
+        # go ahead and build the path to the icon for use by any documents
+        icon_path = os.path.join(
+            self.disk_location,
+            os.pardir,
+            "icons",
+            "photoshop.png"
+        )
+
+        engine = self.parent.engine
+
+        # get the active document name
+        try:
+            active_doc_name = engine.adobe.app.activeDocument.name
```

Comment:
I'd prefer to see a `get_active_document_name` method added to AdobeBridge in tk-photoshopcc. It's probably something we'll be doing often, so constantly having to catch RuntimeError will get annoying.